geoapps/io/constants.py

```diff
@@ -0,0 +1,143 @@
+    "uncertainty_mode",
+    "receivers_offset",
+    "topography",
+    "result_folder",
+    "detrend",
+    "data_file",
+    "new_uncert",
+    "input_mesh",
+    "save_to_geoh5",
+    "input_mesh_file",
+    "inversion_mesh_type",
+    "shift_mesh_z0",
+    "receivers_offset",
+    "gradient_type",
+    "initial_beta",
+    "initial_beta_ratio",
+    "lower_bound",
+    "upper_bound",
+    "max_distance",
+    "max_chunk_size",
+    "chunk_by_rows",
+    "output_tile_files",
+    "no_data_value",
+    "parallelized",
+]
+
+required_parameters = ["inversion_type", "core_cell_size"]
+
+valid_parameter_values = {
+    "inversion_type": ["gravity", "magnetics", "mvi", "mvic"],
+    "inversion_style": ["voxel"],
+    "data_format": ["ubc_grav", "ubc_mag", "GA_object"],
+    "inversion_mesh_type": ["TREE"],
+    "gradient_type": ["total"],
+}
+
+valid_parameter_types = {
+    "inversion_type": [str],
+    "core_cell_size": [int, float],
+    "workpath": [str],
+    "inversion_style": [str],
+    "forward_only": [bool],
+    "result_folder": [str],
+    "inducing_field_aid": [int, float],
+    "resolution": [int, float],
+    "window": [dict],
+    "workspace": [str],
+    "data": [dict],
+    "ignore_values": [str],
+    "detrend": [dict],
+    "data_file": [str],
+    "new_uncert": [int, float],
+    "input_mesh": [str],
+    "save_to_geoh5": [str],
+    "inversion_mesh_type": [str],
+    "shift_mesh_z0": [int, float],
+    "topography": [dict],
+    "receivers_offset": [dict],
+    "chi_factor": [int, float],
+    "model_norms": [int, float],
+    "max_iterations": [int],
+    "max_cg_iterations": [int],
+    "tol_cg": [int, float],
+    "max_global_iterations": [int],
+    "gradient_type": [str],
+    "initial_beta": [int, float],
+    "initial_beta_ratio": [float],
+    "n_cpu": [int, float],
+    "max_ram": [int, float],
+    "depth_core": [dict],
+    "padding_distance": [int, float],
+    "octree_levels_topo": [int, float],
+    "octree_levels_obs": [int, float],
+    "octree_levels_padding": [int, float],
+    "alphas": [int, float],
+    "lower_bound": [int, float],
+    "upper_bound": [int, float],
+    "max_distance": [int, float],
+    "max_chunk_size": [int, float],
+    "chunk_by_rows": [bool],
+    "output_tile_files": [bool],
+    "no_data_value": [int, float],
+    "parallelized": [bool],
+    "out_group": [str],
+    "reference_model": [dict],
+    "starting_model": [dict],
+}
+
+valid_parameter_shapes = {
+    "inducing_field_aid": (3,),
+    "new_uncert": (2,),
+    "padding_distance": (3, 2),
+}
+
+valid_parameter_keys = {
+    "window": ["center_x", "center_y", "width", "height", "azimuth", "center", "size"],
+    "data": ["type", "name", "channels"],
+    "topography": ["GA_object", "drapped", "constant", "file"],
+    "receivers_offset": ["constant", "constant_drape", "radar_drape"],
+    "reference_model": ["model", "value", "none"],
+    "starting_model": ["model", "value"],
+    "depth_core": ["value", "auto"],
+}
```

Comment:
Would you consider changing the storage format as 
`"parameter_X": {
    "type": "val",
    "shape": (*, ),
    "required": True,
    "accepted": ["val_A", "val_B"]
}`
Then the whole validation could be done in a single loop?

Reply:
sure thing!